src/libraries/System.Private.CoreLib/src/System/Reflection/ModifiedFunctionPointerType.cs

```diff
@@ -0,0 +1,114 @@
+                nestedSignatureParameterIndex: 0);
+
+            Type[] parameters = functionPointerType.GetFunctionPointerParameterTypes();
+            int count = parameters.Length;
+            ModifiedType[] modifiedTypes = new ModifiedType[count];
+            for (int i = 0; i < count; i++)
+            {
+                modifiedTypes[i] = Create(
+                    parameters[i],
+                    signatureProvider,
+                    rootSignatureParameterIndex,
+                    nestedSignatureIndex + 1,
+                    nestedSignatureParameterIndex: i + 1);
+            }
+
+            _parameterTypes = modifiedTypes;
+        }
+
+        public override Type GetFunctionPointerReturnType() => _returnType;
+        public override Type[] GetFunctionPointerParameterTypes() => CloneArray<Type>(_parameterTypes);
+
+        public override Type[] GetFunctionPointerCallingConventions()
+        {
+            _callingConventions ??= CreateCallingConventions();
+            return CloneArray(_callingConventions);
+        }
+
+        private Type[] CreateCallingConventions()
+        {
+            Type[] returnTypeOptionalModifiers = GetFunctionPointerReturnType().GetOptionalCustomModifiers();
+
+            ArrayBuilder<Type> builder = default;
+
+            bool foundCallingConvention = false;
+
+            for (int i = 0; i < returnTypeOptionalModifiers.Length; i++)
+            {
+                Type type = returnTypeOptionalModifiers[i];
+                if (type.FullName!.StartsWith(CallingConventionTypePrefix, StringComparison.Ordinal))
+                {
+                    builder.Add(type);
+
+                    if (type == typeof(CallConvCdecl) ||
+                        type == typeof(CallConvFastcall) ||
+                        type == typeof(CallConvStdcall) ||
+                        type == typeof(CallConvThiscall))
+                    {
+                        foundCallingConvention = true;
+                    }
+                }
+            }
+
+            if (!foundCallingConvention)
+            {
+                // Normalize the calling conventions by manufacturing a type.
+                switch (GetCallingConvention())
+                {
+                    case MdSigCallingConvention.C:
+                        builder.Add(typeof(CallConvCdecl));
+                        break;
+                    case MdSigCallingConvention.StdCall:
+                        builder.Add(typeof(CallConvStdcall));
+                        break;
+                    case MdSigCallingConvention.ThisCall:
+                        builder.Add(typeof(CallConvThiscall));
+                        break;
+                    case MdSigCallingConvention.FastCall:
+                        builder.Add(typeof(CallConvFastcall));
+                        break;
+                }
+            }
+
+            return builder.Count == 0 ? EmptyTypes : builder.ToArray();
+        }
```

Comment:
OK thanks. I know we talked about this before, but it wasn't clear to me that it is essentially invalid to have both one of these CallKind flags plus the actual modifier, so this code was trying to protect from that case to prevent the attribute from being added when it was already there.